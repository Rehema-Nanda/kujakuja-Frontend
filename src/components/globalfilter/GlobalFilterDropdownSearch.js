import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { Input } from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import i18n from "i18next";

const GlobalFilterDropdownSearch = ({
    className, items, itemLabelProperty, setResults, searchTerm, setSearchTerm,
}) => {
    const handleSearch = useCallback(
        (e) => {
            const eventSearchTerm = e.target.value;
            const lowerCasedSearchTerm = eventSearchTerm.toLowerCase();
            const results = items.filter((item) => {
                return item[itemLabelProperty].toLowerCase().includes(lowerCasedSearchTerm);
            });
            setResults(results);
            setSearchTerm(eventSearchTerm);
        }, [items, itemLabelProperty, setResults, setSearchTerm],
    );

    return (
        <div className={`${className} globalfilter-search-container`}>
            <div className="map-controls-search-icon">
                <FontAwesomeIcon icon="search" />
            </div>
            <Input
                type="text"
                onChange={handleSearch}
                className="map-controls-search"
                placeholder={i18n.t("globalHeader.search")}
                defaultValue={searchTerm}
            />
        </div>
    );
};

GlobalFilterDropdownSearch.propTypes = {
    className: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    itemLabelProperty: PropTypes.string.isRequired,
    setResults: PropTypes.func.isRequired,
    searchTerm: PropTypes.string,
    setSearchTerm: PropTypes.func,
};

GlobalFilterDropdownSearch.defaultProps = {
    className: "",
    searchTerm: "",
    setSearchTerm: () => {},
};

export default GlobalFilterDropdownSearch;
