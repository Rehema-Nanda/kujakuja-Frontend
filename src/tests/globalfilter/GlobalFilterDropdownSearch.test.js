import React from "react";
import { mount } from "enzyme";
import GlobalFilterDropdownSearch from "../../components/globalfilter/GlobalFilterDropdownSearch";

describe("GlobalFilterDropdownSearch component", () => {
    const props = {
        items: [],
        itemLabelProperty: "name",
        setResults: jest.fn(),
    };
    const wrapper = mount(<GlobalFilterDropdownSearch {...props} />);

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
