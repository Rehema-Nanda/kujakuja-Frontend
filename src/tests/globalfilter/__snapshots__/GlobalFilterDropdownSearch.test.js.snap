// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFilterDropdownSearch component renders correctly 1`] = `
"<GlobalFilterDropdownSearch items={{...}} itemLabelProperty=\\"name\\" setResults={[Function: mockConstructor]} className=\\"\\" searchTerm=\\"\\" setSearchTerm={[Function: setSearchTerm]}>
  <div className=\\" globalfilter-search-container\\">
    <div className=\\"map-controls-search-icon\\">
      <FontAwesomeIcon icon=\\"search\\">
        <i className=\\"fa\\" />
      </FontAwesomeIcon>
    </div>
    <Input type=\\"text\\" onChange={[Function]} className=\\"map-controls-search\\" placeholder={[undefined]} defaultValue=\\"\\">
      <input onChange={[Function]} placeholder={[undefined]} defaultValue=\\"\\" type=\\"text\\" className=\\"map-controls-search form-control\\" />
    </Input>
  </div>
</GlobalFilterDropdownSearch>"
`;
